Slider/Assets/Scripts/Sliders/Grid/AreaGrids/MagiTechGrid.cs

```diff
@@ -81,39 +90,129 @@ public static bool IsInPast(Transform transform)
+        List<Collider2D> list = new();
+        collider.OverlapCollider(filter, list);
+        return list;
+    }
+
     public void HasTwoBurgers(Condition c)
     {
-        if (desyncBurger.IsDesynced)
+        if (!desyncBurger.IsDesynced)
         {
-            BoxCollider2D collider = hungryBoi.GetComponent<BoxCollider2D>();
+            c.SetSpec(false);
+            return;
+        }
 
-            ContactFilter2D filter = new ContactFilter2D();
-            filter.layerMask = LayerMask.GetMask("Item");
-            filter.useLayerMask = true;
+        bool hasBurger = false;
+        bool hasDesyncBurger = false;
 
-            List<Collider2D> list = new List<Collider2D>(); //Change this to however many items can be put inside hungry boi's collider
-            collider.OverlapCollider(filter, list);
+        foreach (Collider2D hit in GetCollidingItems(hungryBoi.gameObject))
+        {
+            if (hit != null)
+            {
+                Item item = hit.GetComponent<Item>();
+                //Debug.Log(item.itemName);
+                if (item.itemName == "Burger") hasBurger = true;
+                else if (item.itemName == desyncBurger.itemName) hasDesyncBurger = true;
+            }
+        }
+        c.SetSpec(hasBurger && hasDesyncBurger);
+    }
 
-            bool hasBurger = false;
-            bool hasDesyncBurger = false;
+    public void FireHasStool(Condition c)
+    {
+        if (gems.GetValueOrDefault(Area.Mountain))
+        {
+            c.SetSpec(true);
+            return;
+        }
+        foreach (Collider2D hit in GetCollidingItems(fireBoi.gameObject))
+        {
+            if (hit != null)
+            {
+                Item item = hit.GetComponent<Item>();
+                if (item.itemName == "Step Stool")
+                {
+                    c.SetSpec(true);
+                    return;
+                }
+            }
+        }
+        c.SetSpec(false);
+    }
+    public void LightningHasStool(Condition c)
+    {
+        if (gems.GetValueOrDefault(Area.Factory))
+        {
+            c.SetSpec(true);
+            return;
+        }
 
-            foreach (Collider2D hit in list)
+        foreach (Collider2D hit in GetCollidingItems(lightningBoi.gameObject))
+        {
+            if (hit != null)
             {
-                if (hit != null)
+                Item item = hit.GetComponent<Item>();
+                if (item.itemName == "Step Stool")
                 {
-                    Item item = hit.GetComponent<Item>();
-                    //Debug.Log(item.itemName);
-                    if (item.itemName == "Burger") hasBurger = true;
-                    else if (item.itemName == desyncBurger.itemName) hasDesyncBurger = true;
+                    c.SetSpec(true);
+                    return;
                 }
             }
-            c.SetSpec(hasBurger && hasDesyncBurger);
         }
-        else
+        c.SetSpec(false);
+    }
+
+    #region Gem Conds
```

Comment:
it might be worth putting all the gem related code in its own `GemManager` script or something, so that magitech grid is mostly puzzle related